src/pint/models/fdjump.py

```diff
@@ -0,0 +1,155 @@
+"""System and frequency dependent delays to model profile evolution."""
+
+import re
+from warnings import warn
+
+import astropy.units as u
+import numpy as np
+
+from pint.models.parameter import boolParameter, maskParameter
+from pint.models.timing_model import DelayComponent
+
+fdjump_max_index = 20
+
+
+class FDJump(DelayComponent):
+    """A timing model for system-dependent frequency evolution of pulsar
+    profiles.
+
+    This model expresses the delay as a polynomial function of the
+    observing frequency/logarithm of observing frequency. This is
+    intended to compensate for the delays introduced by frequency-dependent
+    profile structure when a frequency-independent template profile is used.
```

Comment:
Can you explain the indexing scheme and tempo2 conversion here?

Reply:
Added.